File: Java/Count and Say.java
// 20ms, 10.43%; 42MB, 5.26%
class Solution {
    public String countAndSay(int n) {
        String res = "1";
        while (--n > 0) {
            String tmp = "";
            Character last = res.charAt(0);
            int count = 1;
            for (int i = 1; i < res.length(); ++i) {
                if (res.charAt(i) != last) {
                    tmp += String.valueOf(count) + last;
                    last = res.charAt(i);
                    count = 1;
                } else {
                    count += 1;
                }
            }
            res = tmp + String.valueOf(count) + last;
        }
        return res;
    }
}

File: Go/Count and Say.go
// 0ms, 100.00%; 6.4MB, 50.00%
func countAndSay(n int) string {
	res := "1"
	for i := 1; i < n; i++ {
		tmp, count, last := "", 0, rune(res[0])
		for _, c := range res {
			if c == last {
				count++
			} else {
				tmp += strconv.Itoa(count) + string(last)
				count, last = 1, c
			}
		}
		res = tmp + strconv.Itoa(count) + string(last)
	}
	return res
}
